code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubExternalIdHandler.java

```diff
@@ -77,7 +77,15 @@ public ExternalId untransform(ExternalId externalId) {
         if(StringUtils.isBlank(jwt) && !StringUtils.isBlank(accessToken)){
             try {
                 jwt =  ProjectConstants.AUTH_TYPE + githubTokenHandler.getGithubToken(accessToken).getJwt();
-            } catch (IOException e) {
+            } catch (ClientVisibleException e) {
```

Comment:
The driving motivation for this change is so that if the access token becomes invalid we remove the cached one and the user will have to get a new one. Otherwise the error gets masked by another one and it is not apparent that the token is invalid/out of date. More specifically if you have an access Token and auth switches from GHE to github or vice versa everything blows up.